onnx/defs/math/defs.cc

```diff
@@ -3337,4 +3337,220 @@ ONNX_OPERATOR_SET_SCHEMA(
+
+    if (full_matrices) {
+      // Full SVD
+      // U and Vh are both square matrices
+      *U_shape->add_dim() = A_n_rows_dim;
+      *U_shape->add_dim() = A_n_rows_dim;
+
+      *S_shape->add_dim() = min_dimension;
+
+      *Vh_shape->add_dim() = A_n_cols_dim;
+      *Vh_shape->add_dim() = A_n_cols_dim;
+    } else {
+      // Partial SVD
+      // U's number of columns becomes the min dimension
+      // Vh's number of rows becomes the min dimension
+      *U_shape->add_dim() = min_dimension;
+      *U_shape->add_dim() = A_n_rows_dim;
+
+      *S_shape->add_dim() = min_dimension;
+
+      *Vh_shape->add_dim() = min_dimension;
+      *Vh_shape->add_dim() = A_n_cols_dim;
+    }
+  } else {
+    *S_shape->add_dim() = min_dimension;
+  }
+}
+
+ONNX_OPERATOR_SET_SCHEMA(
+    SVD,
+    18,
+    OpSchema()
+        .SetDoc(SVD_doc)
+        .Input(
+            0,
+            "A",
+            "Input tensor to factor. Must have at least 2 dimensions.",
+            "T",
+            OpSchema::Single,
+            true,
+            1,
+            // TODO
+            OpSchema::Unknown)
+        .Output(
+            0,
+            "U",
+            "Left singular vectors as matrix. See operator doc for dimension information.",
+            "T",
+            OpSchema::Single,
+            true,
+            1,
+            // TODO
+            OpSchema::Unknown)
+        .Output(
+            1,
+            "S",
+            "Vector of singular values. See operator doc for dimension information.",
+            "T",
+            OpSchema::Single,
+            true,
+            1,
+            // TODO
+            OpSchema::Unknown)
+        .Output(
+            2,
+            "Vh",
+            "Right singular vectors as matrix. See operator doc for dimension information.",
+            "T",
+            OpSchema::Single,
+            true,
+            1,
+            // TODO
+            OpSchema::Unknown)
+        .TypeConstraint(
+            "T",
+            OpSchema::all_numeric_types_with_bfloat_and_complex(),
+            "Constrain input and output types to all numeric tensor types")
+        .Attr(
+            "full_matrices",
+            "If true (1), compute the full SVD. If false (0), compute the reduced SVD. Defaults to true (1).",
+            AttributeProto::INT,
+            false)
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) { SVDInferenceFunction(ctx, true); }));
+
+static const char* SVDVals_doc = R"DOC(
+Singular Value Decomposition - only compute the singular values.
+
+See SVD operator doc for general SVD factorization and dimension information.
+SVDVals only computes and outputs S, the vector of singular values.
+
+Full vs partial SVD does not apply as the vector of singular values will always have the same dimensions.
+
+TODO - add doc about differentiability. According to pytorch, svdvals is always differentiable
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
```

Comment:
Why not having an optional output instead of two operators?

Reply:
My intuition was having two separate operators with known outputs instead of one with optional outputs would be easier for consumers and runtimes but happy to change it to one operator with optional outputs! 

As an additional datapoint, pytorch has two separate operators. Tensorflow and numpy have a single operator with optional output.

Reply:
A couple of onnx  operators already have optional outputs. I don't know what the first authors of onnx imagined for this kind of functions but i did not find any other case in onnx with two operators doing the same thing but returning different outputs. Maybe another opinion would be better.

Reply:
Yep if preferring optional outputs is more standard today in onnx, I would prefer to keep it consistent. Would be rather strightforward to combine them in the current PR

I'm not sure who to ping for the other opinion?

Reply:
@gramalingam may be able to help about this topic.

Reply:
@gramalingam any thoughts here? :)

Reply:
Ok great, will update!

Reply:
ok done, combined them :)